src/server/http/short_url_lookup.js

```diff
@@ -0,0 +1,101 @@
+const crypto = require('crypto');
+
+export default function (server) {
+  async function updateMetadata(urlId, urlDoc) {
+    const client = server.plugins.elasticsearch.client;
+
+    try {
+      await client.update({
+        index: '.kibana',
+        type: 'url',
+        id: urlId,
+        body: {
+          doc: {
+            'accessDate': new Date(),
+            'accessCount': urlDoc._source.accessCount + 1
+          }
+        }
+      });
+    } catch (err) {
+      console.log('Warning: Error updating url metadata', err);
+      //swallow errors. It isn't critical if there is no update.
+    }
+  }
+
+  async function getUrlDoc(urlId) {
+    const urlDoc = await new Promise((resolve, reject) => {
+      const client = server.plugins.elasticsearch.client;
+
+      client.get({
+        index: '.kibana',
+        type: 'url',
+        id: urlId
+      })
+      .then(response => {
+        resolve(response);
+      })
+      .catch(err => {
+        resolve();
+      });
+    });
+
+    return urlDoc;
+  }
+
+  async function createUrlDoc(url, urlId) {
+    const newUrlId = await new Promise((resolve, reject) => {
+      const client = server.plugins.elasticsearch.client;
+
+      client.index({
+        index: '.kibana',
+        type: 'url',
+        id: urlId,
+        body: {
+          url,
+          'accessCount': 0,
+          'createDate': new Date(),
+          'accessDate': new Date()
+        }
+      })
+      .then(response => {
+        resolve(response._id);
+      })
+      .catch(err => {
+        reject(err);
+      });
+    });
+
+    return newUrlId;
+  }
+
+  function createUrlId(url) {
+    const urlId = crypto.createHash('md5')
+    .update(url)
+    .digest('hex');
+
+    return urlId;
+  }
+
+  return {
+    async generateUrlId(url) {
+      const urlId = createUrlId(url);
+
+      const urlDoc = await getUrlDoc(urlId);
+      if (urlDoc) return urlId;
+
+      return createUrlDoc(url, urlId);
+    },
+    async getUrl(urlId) {
+      try {
+        const urlDoc = await getUrlDoc(urlId);
+        if (!urlDoc) throw new Error('Requested shortened url does note exist in kibana index');
+
+        updateMetadata(urlId, urlDoc);
+
+        return urlDoc._source.url;
+      } catch (err) {
+        return '/';
```

Comment:
thoughts on letting the server throw instead of redirecting to /?  similar to how url/app/asdf will throw an invalid app error, or other pages give 404s